### PR TITLE
migrations: Add test to ensure extension modifications are marked as privileged

### PR DESCRIPTION
**Pull request by efritz**
Add a verbose error when new migrations altering Postgres extensions are not marked as privileged.

## Test plan

This adds tests that should catch any violations in CI.

### CODE REVIEW

File: internal/database/migration/definition/read.go [hunk 1]
```diff
@@ -201,6 +213,13 @@ func parseIndexMetadata(queryText string) (*IndexMetadata, bool) {
 	}, true
 }
 
+var alterExtensionPattern = lazyregexp.New(`(CREATE|COMMENT ON|DROP) EXTENSION .+;\n*`)
```
Comment (eseliger) [inline internal/database/migration/definition/read.go:216]: should this match case insensitive?

Comment (eseliger) [reply internal/database/migration/definition/read.go:216]: what about more than 1 space, or multi-line statements? Should we cover those as well?

**Review by eseliger — APPROVED** [review]

File: internal/database/migration/definition/read.go [hunk 2]
```diff
@@ -165,6 +165,18 @@ func hydrateMetadataFromFile(fs fs.FS, filepath string, definition Definition) (
 		}
 	}
 
+	if isPrivileged(definition.UpQuery.Query(sqlf.PostgresBindVar)) || isPrivileged(definition.DownQuery.Query(sqlf.PostgresBindVar)) {
+		if !payload.Privileged {
+			return Definition{}, instructionalError{
+				class:       "malformed Postgres extension modification",
+				description: "did not expect queries that require elevated permissions",
+				instructions: strings.Join([]string{
+					"Add `privileged: true` to this migration's metadata.yaml file.",
```
Comment (mrnugget) [inline internal/database/migration/definition/read.go:174]: Idea (non-blocking): would be cool if the file path was included in the error message to make it easier to open.

**Review by mrnugget — APPROVED** [review]

File: internal/database/migration/definition/read.go (hunk 1 above)
Comment (efritz) [reply internal/database/migration/definition/read.go]: Updated.

File: internal/database/migration/definition/read.go (hunk 2 above)
Comment (efritz) [reply internal/database/migration/definition/read.go]: Updated.

**Review by mrnugget — APPROVED** [review]
Nice!

**Pull request merged**